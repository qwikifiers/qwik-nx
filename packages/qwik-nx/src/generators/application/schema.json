{
  "$schema": "http://json-schema.org/schema",
  "$id": "QwikNx",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the application.",
      "pattern": "^[a-zA-Z][^:]*$",
      "x-priority": "important"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "$default": { "$source": "argv", "index": 0 },
      "x-prompt": "Which directory do you want to create the application in?"
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css",
      "alias": "s",
      "x-prompt": {
        "message": "Which stylesheet format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SASS(.scss)       [ http://sass-lang.com                     ]"
          },
          {
            "value": "styl",
            "label": "Stylus(.styl)     [ http://stylus-lang.com                   ]"
          },
          {
            "value": "less",
            "label": "LESS              [ http://lesscss.org                       ]"
          }
        ],
        "default": "css"
      }
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "vitest"
    },
    "strict": {
      "type": "boolean",
      "description": "Creates an application with strict mode and strict type checking.",
      "default": true
    },
    "tailwind": {
      "description": "Setup Tailwind",
      "type": "boolean",
      "default": false
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["playwright", "cypress", "none"],
      "description": "Test runner to use for end to end (E2E) tests.",
      "default": "none"
    },
    "devServerPort": {
      "type": "number",
      "description": "The port for the dev server of the app.",
      "default": 5173
    },
    "previewServerPort": {
      "type": "number",
      "description": "The port for the preview server of the app.",
      "default": 4173
    }
  },
  "required": ["directory"]
}
