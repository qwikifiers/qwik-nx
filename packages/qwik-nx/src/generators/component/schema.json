{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Component",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the component?"
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "alias": "p",
      "$default": {
        "$source": "projectName"
      },
      "x-prompt": "What is the name of the project for this component?"
    },
    "directory": {
      "type": "string",
      "description": "Create the component under this directory (can be nested)."
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css",
      "alias": "s",
      "x-prompt": {
        "message": "Which stylesheet format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SASS(.scss)       [ http://sass-lang.com                     ]"
          },
          {
            "value": "styl",
            "label": "Stylus(.styl)     [ http://stylus-lang.com                   ]"
          },
          {
            "value": "less",
            "label": "LESS              [ http://lesscss.org                       ]"
          },
          {
            "value": "none",
            "label": "none"
          }
        ],
        "default": "css"
      }
    },
    "skipTests": {
      "description": "When true, does not create `spec.ts` test files for the new component.",
      "type": "boolean"
    },
    "flat": {
      "type": "boolean",
      "description": "Create component at the source root rather than its own directory.",
      "default": false
    },
    "generateStories": {
      "description": "Create Storybook stories for the component",
      "type": "boolean"
    },
    "exportDefault": {
      "description": "Whether to use default export instead of named one for the generated component.",
      "type": "boolean",
      "default": false
    }
  },
  "required": ["name", "project"]
}
