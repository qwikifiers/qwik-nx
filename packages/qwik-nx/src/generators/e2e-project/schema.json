{
  "$schema": "http://json-schema.org/schema",
  "$id": "E2eProject",
  "title": "Create an E2E app for a Qwik app",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "The name of the frontend project to test."
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["playwright", "cypress"],
      "description": "Test runner to use for end to end (E2E) tests.",
      "default": "playwright"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    }
  },
  "required": ["project", "e2eTestRunner"]
}
