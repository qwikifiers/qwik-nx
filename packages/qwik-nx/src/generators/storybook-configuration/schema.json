{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "StorybookConfiguration",
  "title": "Adds Storybook configuration to a project.",
  "description": "Adds Storybook configuration to a project to be able to use and create stories.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "aliases": ["project", "projectName"],
      "description": "Project for which to generate Storybook configuration.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "For which project do you want to generate Storybook configuration?",
      "x-dropdown": "projects",
      "x-priority": "important"
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "js": {
      "type": "boolean",
      "description": "Generate JavaScript story files rather than TypeScript story files.",
      "default": false
    },
    "tsConfiguration": {
      "type": "boolean",
      "description": "Configure your project with TypeScript. Generate main.ts and preview.ts files, instead of main.js and preview.js.",
      "default": true,
      "x-priority": "important"
    },
    "qwikCitySupport": {
      "type": "string",
      "enum": ["true", "false", "auto"],
      "x-priority": "important",
      "description": "Whether to add support for the qwik city. By default, it will be added for applications, not libraries.",
      "default": "auto"
    }
  },
  "required": ["name"]
}
