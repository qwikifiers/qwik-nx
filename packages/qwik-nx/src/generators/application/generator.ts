import {
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  names,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import { NormalizedSchema, normalizeOptions } from './utils/normalize-options';
import { viteConfigurationGenerator } from '@nrwl/vite';
import { QwikAppGeneratorSchema } from './schema';
import { Linter } from '@nrwl/linter';
import { addStyledModuleDependencies } from '../../utils/add-styled-dependencies';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import { configureEslint } from '../../utils/configure-eslint';
import { addCommonQwikDependencies } from '../../utils/add-common-qwik-dependencies';

function addFiles(tree: Tree, options: NormalizedSchema) {
  const templateOptions = {
    ...options,
    ...names(options.name),
    template: '',
  };
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.appProjectRoot,
    templateOptions
  );
}

async function configureVite(
  tree: Tree,
  options: NormalizedSchema
): Promise<GeneratorCallback> {
  const tsConfigPath = joinPathFragments(
    options.appProjectRoot,
    'tsconfig.json'
  );
  const tsConfig = tree.read(tsConfigPath);
  const includeVitest = options.unitTestRunner === 'vitest';

  const callback = await viteConfigurationGenerator(tree, {
    project: options.projectName,
    newProject: false,
    includeVitest,
    includeLib: false,
    uiFramework: 'none',
    inSourceTests: false,
  });
  tree.delete(joinPathFragments(options.appProjectRoot, 'index.html'));

  // overwrite tsconfig back after "viteConfigurationGenerator"
  tree.write(tsConfigPath, tsConfig);

  if (includeVitest) {
    const projectConfig = readProjectConfiguration(tree, options.projectName);
    const testTarget = projectConfig.targets['test'];
    testTarget.outputs = ['{workspaceRoot}/coverage/{projectRoot}'];
    updateProjectConfiguration(tree, options.projectName, projectConfig);
  }

  return callback;
}

export default async function (tree: Tree, options: QwikAppGeneratorSchema) {
  const normalizedOptions = normalizeOptions(tree, options);
  const tasks: GeneratorCallback[] = [];

  addProjectConfiguration(tree, normalizedOptions.projectName, {
    root: normalizedOptions.appProjectRoot,
    name: normalizedOptions.projectName,
    projectType: 'application',
    sourceRoot: `${normalizedOptions.appProjectRoot}/src`,
    // TODO: revisit later, now "viteConfigurationGenerator" will fail if there's no build target specified
    targets: { build: { options: {} } },
    tags: normalizedOptions.parsedTags,
  });

  addFiles(tree, normalizedOptions);

  tasks.push(await configureVite(tree, normalizedOptions));

  if (normalizedOptions.linter === Linter.EsLint) {
    tasks.push(configureEslint(tree, normalizedOptions.projectName, true));
  }

  if (normalizedOptions.style !== 'none') {
    tasks.push(
      addStyledModuleDependencies(tree, normalizedOptions.styleExtension)
    );
  }

  tasks.push(addCommonQwikDependencies(tree));

  if (!options.skipFormat) {
    await formatFiles(tree);
  }

  return runTasksInSerial(...tasks);
}
