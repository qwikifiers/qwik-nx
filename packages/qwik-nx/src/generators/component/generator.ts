import {
  formatFiles,
  generateFiles,
  GeneratorCallback,
  getProjects,
  joinPathFragments,
  logger,
  names,
  Tree,
} from '@nrwl/devkit';
import { addStyledModuleDependencies } from '../../utils/add-styled-dependencies';
import { ComponentGeneratorSchema } from './schema';

interface NormalizedSchema extends ComponentGeneratorSchema {
  directory: string;
  hasStyles: boolean;
  projectRoot: string;
}

function getDirectory(host: Tree, options: ComponentGeneratorSchema) {
  const workspace = getProjects(host);
  let baseDir: string;
  if (options.directory) {
    baseDir = options.directory;
  } else {
    baseDir =
      workspace.get(options.project)!.projectType === 'application'
        ? 'components'
        : 'lib';
  }
  return options.flat
    ? baseDir
    : joinPathFragments(baseDir, names(options.name).fileName);
}

function normalizeOptions(
  host: Tree,
  options: ComponentGeneratorSchema
): NormalizedSchema {
  const project = getProjects(host).get(options.project);

  if (!project) {
    logger.error(
      `Cannot find the ${options.project} project. Please double check the project name.`
    );
    throw new Error();
  }

  const projectRoot =
    project.sourceRoot ?? joinPathFragments(project.root, 'src');

  const directory = getDirectory(host, options);

  return {
    ...options,
    directory,
    hasStyles: options.style !== 'none',
    projectRoot,
  };
}

function createComponentFiles(tree: Tree, options: NormalizedSchema) {
  const libNames = names(options.name);
  const hasStyles = options.style && options.style !== 'none';
  const templateOptions = {
    ...options,
    ...libNames,
    hasStyles,
  };

  const componentDir = joinPathFragments(
    options.projectRoot,
    options.directory
  );

  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files/common'),
    componentDir,
    templateOptions
  );
  if (hasStyles) {
    generateFiles(
      tree,
      joinPathFragments(__dirname, 'files/styles'),
      componentDir,
      templateOptions
    );
  }
  if (!options.skipTests) {
    generateFiles(
      tree,
      joinPathFragments(__dirname, 'files/tests'),
      componentDir,
      templateOptions
    );
  }
}

export async function componentGenerator(
  tree: Tree,
  options: ComponentGeneratorSchema
): Promise<GeneratorCallback> {
  const normalizedOptions = normalizeOptions(tree, options);
  createComponentFiles(tree, normalizedOptions);
  await formatFiles(tree);

  return addStyledModuleDependencies(tree, normalizedOptions.style);
}

export default componentGenerator;
