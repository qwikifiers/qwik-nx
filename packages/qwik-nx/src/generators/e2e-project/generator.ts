import {
  ensurePackage,
  GeneratorCallback,
  readProjectConfiguration,
  Tree,
} from '@nrwl/devkit';
import { getInstalledNxVersion } from '../../utils/get-installed-nx-version';
import { nxKitVersion } from '../../utils/versions';
import { E2eProjectGeneratorSchema, NormalizedSchema } from './schema';
import { normalizeOptions } from './utils/normalize-options';

export async function addE2eProject(
  tree: Tree,
  options: E2eProjectGeneratorSchema
): Promise<GeneratorCallback> {
  const projectConfiguration = readProjectConfiguration(tree, options.project);

  if (projectConfiguration.projectType !== 'application') {
    throw new Error('Cannot setup e2e project for the given frontend project.');
  }

  const normalizedOptions = normalizeOptions(tree, options);
  if (options.e2eTestRunner === 'cypress') {
    return addCypress(tree, normalizedOptions);
  }

  if (options.e2eTestRunner === 'playwright') {
    return addPlaywright(tree, normalizedOptions);
  }

  return () => void 0;
}

async function addCypress(tree: Tree, options: NormalizedSchema) {
  await ensurePackage(tree, '@nrwl/cypress', getInstalledNxVersion(tree));
  const { cypressProjectGenerator } = await import('@nrwl/cypress');

  return await cypressProjectGenerator(tree, {
    ...options,
    name: options.e2eProjectName,
    directory: options.directory,
    project: options.project,
    bundler: 'vite',
    skipFormat: options.skipFormat,
  });
}

async function addPlaywright(tree: Tree, options: NormalizedSchema) {
  await ensurePackage(tree, '@nxkit/playwright', nxKitVersion);
  const { projectGenerator } = await import('@nxkit/playwright');

  return await projectGenerator(tree, {
    ...options,
    name: options.e2eProjectName,
    directory: options.directory,
    frontendProject: options.project,
    skipFormat: options.skipFormat,
  });
}

export default addE2eProject;
