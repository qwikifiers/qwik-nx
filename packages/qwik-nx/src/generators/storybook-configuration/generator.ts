import {
  addDependenciesToPackageJson,
  ensurePackage,
  formatFiles,
  generateFiles,
  GeneratorCallback,
  names,
  offsetFromRoot,
  ProjectConfiguration,
  readProjectConfiguration,
  Tree,
} from '@nx/devkit';
import { Linter } from '@nx/linter';
import * as path from 'path';
import {
  ensureMdxTypeInTsConfig,
  ensureRootTsxExists,
} from '../../utils/ensure-file-utils';
import { getInstalledNxVersion } from '../../utils/get-installed-nx-version';
import {
  storybookFrameworkQwikVersion,
  reactDOMVersion,
  reactVersion,
  typesMdx,
} from '../../utils/versions';
import {
  NormalizedSchema,
  StorybookConfigurationGeneratorSchema,
} from './schema';

function addFiles(
  tree: Tree,
  options: StorybookConfigurationGeneratorSchema,
  { root }: ProjectConfiguration
) {
  tree.delete(path.join(root, '.storybook/main.js'));

  const templateOptions = {
    ...options,
    ...names(options.name),
    offsetFromRoot: offsetFromRoot(root),
    projectRoot: root,
    configExtension: options.tsConfiguration ? 'ts' : 'js',
  };
  generateFiles(tree, path.join(__dirname, 'files'), root, templateOptions);

  ensureRootTsxExists(tree, options.name);
  ensureMdxTypeInTsConfig(tree, options.name);
}

function normalizeOptions(
  options: StorybookConfigurationGeneratorSchema,
  projectConfig: ProjectConfiguration
): NormalizedSchema {
  let qwikCitySupportInternal: boolean;
  if (options.qwikCitySupport === 'auto' || !options.qwikCitySupport) {
    // "auto"
    qwikCitySupportInternal = projectConfig.projectType === 'application';
  } else {
    // "true" or "false"
    qwikCitySupportInternal = options.qwikCitySupport === 'true';
  }

  return {
    ...options,
    js: !!options.js,
    linter: options.linter ?? Linter.EsLint,
    tsConfiguration: options.tsConfiguration ?? true,
    qwikCitySupportInternal,
  };
}

export async function storybookConfigurationGenerator(
  tree: Tree,
  options: StorybookConfigurationGeneratorSchema
): Promise<GeneratorCallback> {
  const projectConfig = readProjectConfiguration(tree, options.name);

  const normalizedOptions = normalizeOptions(options, projectConfig);
  const nxVersion = getInstalledNxVersion(tree);

  ensurePackage('@nx/storybook', nxVersion);
  const { configurationGenerator } = await import('@nx/storybook');

  await configurationGenerator(tree, {
    storybook7UiFramework: '@storybook/html-webpack5',
    uiFramework: '@storybook/html',
    bundler: 'vite',
    name: normalizedOptions.name,
    js: normalizedOptions.js,
    linter: normalizedOptions.linter,
    tsConfiguration: normalizedOptions.tsConfiguration,
    storybook7Configuration: true,
    configureCypress: false,
  });

  addFiles(tree, normalizedOptions, projectConfig);
  await formatFiles(tree);

  return addStorybookDependencies(tree);
}

async function addStorybookDependencies(
  tree: Tree
): Promise<GeneratorCallback> {
  const { storybook7Version } = await import(
    '@nx/storybook/src/utils/versions'
  );

  return addDependenciesToPackageJson(
    tree,
    {},
    {
      'storybook-framework-qwik': storybookFrameworkQwikVersion,
      '@storybook/builder-vite': storybook7Version,
      '@storybook/addon-docs': storybook7Version,
      react: reactVersion,
      'react-dom': reactDOMVersion,
      '@types/mdx': typesMdx,
    }
  );
}

export default storybookConfigurationGenerator;
