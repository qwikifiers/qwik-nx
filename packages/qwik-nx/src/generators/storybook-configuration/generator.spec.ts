import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { Tree, readProjectConfiguration } from '@nrwl/devkit';

import { storybookConfigurationGenerator } from './generator';
import { StorybookConfigurationGeneratorSchema } from './schema';
import appGenerator from '../application/generator';
import { Linter } from '@nrwl/linter';
import { libraryGenerator } from '../library/generator';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const devkit = require('@nrwl/devkit');
const getInstalledNxVersionModule = require('../../utils/get-installed-nx-version');

describe('storybook-configuration generator', () => {
  let appTree: Tree;
  const projectName = 'test-project';
  const options: StorybookConfigurationGeneratorSchema = { name: projectName };

  jest.spyOn(devkit, 'ensurePackage').mockReturnValue(Promise.resolve());
  jest
    .spyOn(getInstalledNxVersionModule, 'getInstalledNxVersion')
    .mockReturnValue('15.6.0');

  beforeEach(async () => {
    appTree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

    await appGenerator(appTree, {
      name: projectName,
      e2eTestRunner: 'none',
      linter: Linter.None,
      skipFormat: false,
      strict: true,
      style: 'css',
      unitTestRunner: 'none',
    });
  });

  it('should add required targets', async () => {
    await storybookConfigurationGenerator(appTree, options);
    const config = readProjectConfiguration(appTree, projectName);
    expect(config.targets!['storybook']).toEqual({
      executor: '@nrwl/storybook:storybook',
      options: {
        port: 4400,
        configDir: `apps/${projectName}/.storybook`,
      },
      configurations: {
        ci: {
          quiet: true,
        },
      },
    });
    expect(config.targets!['build-storybook']).toEqual({
      executor: '@nrwl/storybook:build',
      outputs: ['{options.outputDir}'],
      options: {
        configDir: `apps/${projectName}/.storybook`,
        outputDir: `dist/storybook/${projectName}`,
      },
      configurations: {
        ci: {
          quiet: true,
        },
      },
    });

    expect(
      [...appTree.listChanges()]
        .sort((a, b) => a.path.localeCompare(b.path))
        .map((c) => ({ path: c.path, type: c.type }))
    ).toMatchSnapshot();
    expect(
      appTree.read(`apps/${projectName}/.storybook/main.ts`)?.toString()
    ).toMatchSnapshot();
    expect(
      appTree.read(`apps/${projectName}/.storybook/preview.ts`)?.toString()
    ).toMatchSnapshot();
  });

  describe('should conditionally add qwikCity decorator to preview.ts', () => {
    const libProjectName = 'test-project-lib';

    beforeEach(async () => {
      await libraryGenerator(appTree, {
        name: libProjectName,
        linter: Linter.None,
        skipFormat: true,
        strict: true,
        style: 'css',
        unitTestRunner: 'none',
      });
    });

    test.each`
      qwikCitySupport | project
      ${'true'}       | ${libProjectName}
      ${'false'}      | ${libProjectName}
      ${'auto'}       | ${libProjectName}
      ${'true'}       | ${projectName}
      ${'false'}      | ${projectName}
      ${'auto'}       | ${projectName}
    `(
      'matches the snapshot when qwikCitySupport is "$qwikCitySupport" and project is "$project"',
      async ({ qwikCitySupport, project }) => {
        await storybookConfigurationGenerator(appTree, {
          ...options,
          qwikCitySupport,
          name: project,
        });
        expect(
          appTree
            .read(
              `${
                project === libProjectName ? 'libs' : 'apps'
              }/${project}/.storybook/preview.ts`
            )
            ?.toString()
        ).toMatchSnapshot();
      }
    );
  });
});
