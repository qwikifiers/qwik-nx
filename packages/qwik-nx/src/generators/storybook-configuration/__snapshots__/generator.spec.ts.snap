// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storybook-configuration generator should add required targets 1`] = `
Array [
  Object {
    "path": ".prettierignore",
    "type": "CREATE",
  },
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/.eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/.prettierignore",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/.storybook/main.ts",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/.storybook/preview.ts",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/.storybook/tsconfig.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/package.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/public/favicon.svg",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/public/manifest.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/public/robots.txt",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/README.md",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/components/header/header.css",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/components/header/header.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/components/icons/qwik.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/components/router-head/router-head.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/entry.dev.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/entry.preview.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/entry.ssr.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/global.css",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/root.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/routes/flower/flower.css",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/routes/flower/index.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/routes/index.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/routes/layout.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/src/routes/service-worker.ts",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/tsconfig.app.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/tsconfig.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/tsconfig.spec.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/test-project/vite.config.ts",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;
